x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type.ts

```diff
@@ -0,0 +1,303 @@
+  EsQueryAlertState,
+} from './alert_type_params';
+import { STACK_ALERTS_FEATURE_ID } from '../../../common';
+import { ComparatorFns, getHumanReadableComparator } from '../lib';
+import { parseDuration } from '../../../../alerts/server';
+import { buildSortedEventsQuery } from '../../../common/build_sorted_events_query';
+
+export const ES_QUERY_ID = '.es-query';
+
+const DEFAULT_MAX_HITS_PER_EXECUTION = 1000;
+
+const ActionGroupId = 'query matched';
+const ConditionMetAlertInstanceId = 'query matched';
+
+export function getAlertType(
+  logger: Logger
+): AlertType<EsQueryAlertParams, EsQueryAlertState, {}, ActionContext, typeof ActionGroupId> {
+  const alertTypeName = i18n.translate('xpack.stackAlerts.esQuery.alertTypeTitle', {
+    defaultMessage: 'ES query',
+  });
+
+  const actionGroupName = i18n.translate('xpack.stackAlerts.esQuery.actionGroupThresholdMetTitle', {
+    defaultMessage: 'Query matched',
+  });
+
+  const actionVariableContextDateLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextDateLabel',
+    {
+      defaultMessage: 'The date the alert met the threshold condition.',
+    }
+  );
+
+  const actionVariableContextValueLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextValueLabel',
+    {
+      defaultMessage: 'The value that met the threshold condition.',
+    }
+  );
+
+  const actionVariableContextHitsLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextHitsLabel',
+    {
+      defaultMessage: 'The documents that met the threshold condition.',
+    }
+  );
+
+  const actionVariableContextMessageLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextMessageLabel',
+    {
+      defaultMessage: 'A pre-constructed message for the alert.',
+    }
+  );
+
+  const actionVariableContextTitleLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextTitleLabel',
+    {
+      defaultMessage: 'A pre-constructed title for the alert.',
+    }
+  );
+
+  const actionVariableContextIndexLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextIndexLabel',
+    {
+      defaultMessage: 'The index the query was run against.',
+    }
+  );
+
+  const actionVariableContextQueryLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextQueryLabel',
+    {
+      defaultMessage: 'The stringifed ES query.',
+    }
+  );
+
+  const actionVariableContextThresholdLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextThresholdLabel',
+    {
+      defaultMessage:
+        "An array of values to use as the threshold; 'between' and 'notBetween' require two values, the others require one.",
+    }
+  );
+
+  const actionVariableContextThresholdComparatorLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextThresholdComparatorLabel',
+    {
+      defaultMessage: 'A comparison function to use to determine if the threshold as been met.',
```

Comment:
A function to determine if the threshold has been met.

Be sure to change "as" to "has"